subsys/net/l2/openthread/openthread.c

```diff
@@ -424,32 +430,35 @@ static int openthread_init(struct net_if *iface)
 
 	__ASSERT(ot_context->instance, "OT instance is NULL");
 
+//	if (!IS_ENABLED(CONFIG_OPENTHREAD_RCP)) { // TODO: check whether required
```

Comment:
So far we've just disabled SHELL in the project file manually. I think that a nice improvement would be to actually make `OPENTHREAD_SHELL` dependant on `!OPENTHREAD_NCS & !OPENTHREAD_RCP`, it makes little sense to enable shell for these.

Reply:
Will do that on the upstream PR.